methods/guest/src/main.rs

```diff
@@ -18,14 +18,19 @@ use shakmaty::{fen::Fen, san::San, CastlingMode, Chess, FromSetup, Move, Positio
 
 fn main() {
     let inputs: Inputs = env::read();
-    let mv: String = inputs.mv;
+    let moves: Vec<String> = inputs.moves;
     let initial_state: String = inputs.board;
     env::commit(&initial_state);
+    // This is optional to commit:
+    env::commit(&moves.len());
```

Comment:
We _do not have to_ publicly `commit` to total moves.
As we also don't commit the moves, if the amount of moves is hidden we can't tell who wins with a proven check mate!

(Caveat: proving time is proportional to moves, so a [timing attack](https://en.wikipedia.org/wiki/Timing_attack) on the prover could reveal `moves.len()`.)